From cdd9043580e0f0a6c60c21b955f1966c78b2098b Mon Sep 17 00:00:00 2001
From: Karthik Ramanan <a0393906@ti.com>
Date: Mon, 2 Dec 2013 15:52:41 +0530
Subject: [PATCH] input source: prevent input disable during repaint

Under certain race conditions, the input source was getting
disabled which was preventing the evdev events from keyboard or
mouse to reach Weston. The side effect of this issue was
that after the screen gets locked, it was not possible to
come to Weston desktop without killing Weston.

Signed-off-by: Karthik Ramanan <a0393906@ti.com>
---
 src/compositor.c |    4 ----
 1 files changed, 0 insertions(+), 4 deletions(-)

diff --git a/src/compositor.c b/src/compositor.c
index 0e0835f..ced3883 100644
--- a/src/compositor.c
+++ b/src/compositor.c
@@ -1084,10 +1084,6 @@ weston_output_schedule_repaint(struct weston_output *output)
 	wl_event_loop_add_idle(loop, idle_repaint, output);
 	output->repaint_scheduled = 1;
 
-	if (compositor->input_loop_source) {
-		wl_event_source_remove(compositor->input_loop_source);
-		compositor->input_loop_source = NULL;
-	}
 }
 
 WL_EXPORT void
-- 
1.7.0.4

